samples/cxx/custom/custom.cpp

```diff
@@ -0,0 +1,197 @@
+/* @file custom.cpp
+ * Solve a closed-system constant pressure ignition problem where the governing equations
+ * are custom-implemented.
+ */
+
+#include "cantera/thermo.h"
+#include "cantera/kinetics.h"
+#include "cantera/numerics/Integrator.h"
+#include <fstream>
+
+using namespace Cantera;
+
+class ReactorODEs : public FuncEval {
+public:
+    /**
+     * Constructor
+     * @param[in] sol Solution object specifying initial system state.
+     */
+    ReactorODEs(shared_ptr<Solution> sol) {
+        /* ---------------------- INITIALIZE MEMBER VARS ---------------------- */
+
+        // pointer to the system's ThermoPhase object. updated by the solver during
+        // simulation to provide iteration-specific thermodynamic properties.
+        gas = sol->thermo();
+
+        // pointer to the kinetics manager. provides iteration-specific species
+        // production rates based on the current state of the ThermoPhase.
+        kinetics = sol->kinetics();
+
+        // the system's constant pressure, taken from the provided initial state.
+        pressure = gas->pressure();
+
+        // number of chemical species in the system.
+        nSpecies = gas->nSpecies();
+
+        // resize the vector_fp storage containers for species partial molar enthalpies
+        // and net production rates. internal values are updated and used by the solver
+        // per iteration.
+        hbar.resize(nSpecies);
+        wdot.resize(nSpecies);
+
+        // number of equations in the ODE system. a conservation equation for each
+        // species, plus a single energy conservation equation for the system.
+        nEqs = nSpecies + 1;
+    }
+
+    /**
+     * Evaluate the ODE right-hand-side function, ydot = f(t,y).
+     *   - overrided from FuncEval, called by the integrator during simulation.
```

Comment:
```suggestion
     *   - overridden from FuncEval, called by the integrator during simulation.
```
This appears in several locations.